idb/postgres/postgres.go

```diff
@@ -314,10 +332,33 @@ func (db *IndexerDb) AddBlock(block *bookkeeping.Block) error {
 			}
 			metrics.PostgresEvalTimeSeconds.Observe(time.Since(start).Seconds())
 
-			err = writer.AddBlock(block, modifiedTxns, delta)
+			var err1 error
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+
+				f := func(tx pgx.Tx) error {
+					return writer.AddTransactions(block, modifiedTxns, tx)
+				}
+				err1 = db.txWithRetry(serializable, f)
+			}()
+
+			err = w.AddBlock(block, modifiedTxns, delta)
 			if err != nil {
 				return fmt.Errorf("AddBlock() err: %w", err)
 			}
+
+			wg.Wait()
+			isUniqueViolationFunc := func(err error) bool {
+				var pgerr *pgconn.PgError
+				return errors.As(err, &pgerr) && (pgerr.Code == pgerrcode.UniqueViolation)
+			}
+			if (err0 != nil) && !isUniqueViolationFunc(err0) {
+				return fmt.Errorf("AddBlock() err0: %w", err0)
+			}
+			if (err1 != nil) && !isUniqueViolationFunc(err1) {
+				return fmt.Errorf("AddBlock() err1: %w", err1)
+			}
```

Comment:
More descriptive error messages. Also above `s/err/block header`
```suggestion
			if (err0 != nil) && !isUniqueViolationFunc(err0) {
				return fmt.Errorf("AddBlock() participation: %w", err0)
			}
			if (err1 != nil) && !isUniqueViolationFunc(err1) {
				return fmt.Errorf("AddBlock() transactions: %w", err1)
			}
```

Reply:
`err0` and `err1` will tell where they come from. The pattern I've been using is `variablename: variablevalue`.